### PR TITLE
Adds holo-targetting Vulture rounds

### PR DESCRIPTION
**Pull request by VileBeggar**
# About the pull request

Adds holo-targetting rounds for the Vulture. Their characteristics are:

- They deal SIGNIFICANTLY LESS damage, down from 400 to 60.
- They cannot penetrate walls nor mobs.
- They do not slow down mobs that are hit.
- When they hit a target, they will "mark them for death", inflicting a holo stack value of 333. This means that they will take 33% more damage from all sources. A balloon alert will be shown to all nearby mobs indicating the mark, along with the holo-target aura itself.
- **They have IFF.**

image

The bullet increases the target's max capacity for holo stacks increases by 233 (to 333 in total). Any holo-targetting rounds that hit the mob will be able utilize this cap increase. The holo stacks will completely deplete after 35-ish seconds from the initial impact. It takes 5 seconds for the stack drain to begin, with every second passing after that removing 10 stacks. Other holo-targetting rounds will not modify the rate of stack depletion, keeping it at a firm 10 stacks per second. They will, however, stop the drain for 5 seconds.

image

# Explain why it's good for the game

This gun is really fun for the users, encouraging a buddy system that I feel the game desperately needs more of. I will admit that being hit with 400 damage, 50 AP from 2 screens away is not engaging for the opposing side, giving them no counterplay at all.

With these bullets, the Vulture can take a supporting role for the marines instead of being an off-screen death cannon. 33% is a decent damage increase, but it requires multiple marines around the target to truly take advantage of it. Compared to the payoff that a regular Vulture round gives, the lack of IFF for those bullets is understandable. However, given that you NEED friendly troops to get any use out of these rounds, I've given the rounds IFF (with Zonespace's blessing) to broaden the areas and situations where it could get used.

The ammo is still limited as ever, so the sniper team will have to pick their shots. Given how weak the rounds are in comparison to the standard ones, they should rarely be able to snipe off targets on their own (in a rare circumstance that they spot a half-crit xeno, an M4RA with a 4x scope can finish them off just the same). 

Is it weak? Too situational? Maybe, but the weapon is very interesting mechanically and I would just like to see it in play a bit more.

The weapon is still ADMIN-ONLY. This will only give the admins a less blunt variant of the Vulture if they choose to spawn in it.

# Testing Photographs and Procedure
<details>
<summary>Screenshots & Videos</summary>

How being hit feels:

https://github.com/cmss13-devs/cmss13/assets/17518895/a576629e-7961-4421-9932-d707e12477f1

General test:

https://github.com/cmss13-devs/cmss13/assets/17518895/e2fbb117-1cf9-4945-b87a-d63e68f62a10

Bonus damage testing. The M41A does a base damage of 44. A holo-stack of 333 will result in roughly 44 * 1.33 = 59 damage.

image

</details>

# Changelog
:cl:
add: The Vulture can now be loaded with holo-targetting rounds which have severely hampered ballistic performance, but mark any target that is hit, increasing their damage taken by 33% for a brief period of time.
/:cl:

### CODE REVIEW

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

File: code/datums/components/bonus_damage_stack.dm [hunk 1]
```diff
@@ -15,10 +15,15 @@
 	var/bonus_damage_cap = 100
 	/// Last world.time that the afflicted was hit by a holo-targeting round.
 	var/last_stack
+	/// extra cap limit added by more powerful bullets
+	var/bonus_damage_cap_increase = 0
+	var/stack_loss_multiplier = 1
 
-/datum/component/bonus_damage_stack/Initialize(bonus_damage_stacks, time)
+/datum/component/bonus_damage_stack/Initialize(bonus_damage_stacks, time, bonus_damage_cap_increase, stack_loss_multiplier)
```
Comment (Zonespace27) [inline code/datums/components/bonus_damage_stack.dm:22]: Did you make sure to change all the addcomponent calls of this component?

**Review by Zonespace27 — COMMENTED** [review]
Interesting concept

**Comment by cm13-github** [comment]
This pull request has conflicts, please resolve those before we can evaluate the pull request.

File: code/datums/ammo/bullet/rifle.dm [hunk 2]
```diff
@@ -22,10 +22,12 @@
 	name = "holo-targeting rifle bullet"
 	damage = 30
 	var/holo_stacks = 10
+	var/bonus_damage_cap_increase = 0
+	var/stack_loss_multiplier = 1
```
Comment (Zonespace27) [inline code/datums/ammo/bullet/rifle.dm:26]: code doc

File: code/datums/ammo/bullet/sniper.dm [hunk 3]
```diff
@@ -139,6 +139,25 @@
 		BULLET_TRAIT_ENTRY(/datum/element/bullet_trait_penetrating/heavy)
 	))
 
+/datum/ammo/bullet/sniper/anti_materiel/vulture/holo_target
+	name = "holo-targeting anti-materiel sniper bullet"
+	damage = 60 // it's a big bullet but its purpose is to support marines, not to kill enemies by itself
+	var/holo_stacks = 333 // inflicts max holo stacks in one hit. this equates to a 33% damage bonus
+	var/bonus_damage_cap_increase = 233 // increases the default holo stack cap of 100 by 233, to 333 in total.
+	var/stack_loss_multiplier = 2 // multiplies the default holo stack drain of 5 stacks per second to 10
```
Comment (Zonespace27) [inline code/datums/ammo/bullet/sniper.dm:147]: code doc

File: code/datums/ammo/bullet/special_ammo.dm [hunk 4]
```diff
@@ -48,10 +48,12 @@
 	damage = 30
 	///Stuff for the HRP holotargetting stacks
 	var/holo_stacks = 15
+	var/bonus_damage_cap_increase = 1
+	var/stack_loss_multiplier = 1
```
Comment (Zonespace27) [inline code/datums/ammo/bullet/special_ammo.dm:52]: .

File: code/datums/ammo/shrapnel.dm [hunk 5]
```diff
@@ -43,10 +43,14 @@
 	shrapnel_chance = 0
 	shell_speed = AMMO_SPEED_TIER_3//she fast af boi
 	penetration = ARMOR_PENETRATION_TIER_5
+	var/holo_stacks = 10
+	var/bonus_damage_cap_increase = 1
+	var/stack_loss_multiplier = 1
```
Comment (Zonespace27) [inline code/datums/ammo/shrapnel.dm:48]: .

**Review by Zonespace27 — CHANGES_REQUESTED** [review]

**Comment by cm13-github** [comment]
Conflicts have been resolved. A maintainer will review the pull request shortly.

File: code/datums/ammo/bullet/rifle.dm [hunk 6]
```diff
@@ -21,11 +21,13 @@
 /datum/ammo/bullet/rifle/holo_target
 	name = "holo-targeting rifle bullet"
 	damage = 30
-	var/holo_stacks = 10
+	var/holo_stacks = 10 /// inflicts this many holo stacks per bullet hit
```
Comment (Zonespace27) [inline code/datums/ammo/bullet/rifle.dm:24]:
```
/// code doc needs to be like this
var/foo /// not like this
```

**Review by Zonespace27 — APPROVED** [review]
Might need some tweaking, but this is an admin item so it's not worth a TM

**Pull request merged**